bayesian/utils.py

```diff
@@ -337,9 +342,13 @@ def build_nested_schema_dict(schema_dict):
 
 def get_next_component_from_graph(ecosystem, user_id):
     tag_count = 2
-    qstring = "user = g.V().has('userid', '{uid}').next();\
-            g.V(user).as('u').V().has('ecosystem','{e}').has('manual_tagging_required', true).has('tags_count', not(within({tc}))).coalesce(inE('has_tagged').as('pkg').outV().as('b').where('u', neq('b')).outE('has_tagged').inV().as('np').where('pkg', neq('np')), V().has('ecosystem','{e}').has('manual_tagging_required', true).not(inE('has_tagged'))).limit(1).valueMap();".format(
-                    uid=user_id, e=ecosystem, tc=tag_count)
+    qstring = ("user = g.V().has('userid', '{uid}').next();"
```

Comment:
@ravsa - on second thoughts, I guess, its better to create a user node while fetching. You can implement the way it is done below
`"user = g.V().has('userid','{user_id}').tryNext().orElseGet{{graph.addVertex("
               "'vertex_label', 'User', 'userid', '{user_id}', {company})}};
`